### PR TITLE
Add support for `Settings` in `Neo4jStore` (fixes for latest `gufe`)

### PR DESCRIPTION
**Pull request by dotsdl**
- Added `Settings` support to state store
- Fixes to allow for presence of Settings in serialized/deserialized bits

### CODE REVIEW

**Comment by dotsdl** [comment]
Ready for review! @mikemhenry and @hmacdope, whichever one of you wants to review, I'd be most grateful!

File: alchemiscale/base/client.py [hunk 1]
```diff
@@ -119,9 +118,9 @@ def _post_resource(self, resource, data):
         return resp.json()
 
     def get_info(self):
-        return self._get_resource("/info", params={}, return_gufe=False)
+        return self._get_resource("/info")
 
     @_use_token
     def _api_check(self):
         # Check if the API is up and running and can reach services
-        self._get_resource("/check", params={}, return_gufe=False)
```
Comment (hmacdope) [inline alchemiscale/base/client.py:127]: I think this is a good idea moving the gufe logic outside this method, I kept getting caught out by it.

File: alchemiscale/storage/statestore.py [hunk 2]
```diff
@@ -319,6 +320,11 @@ def _gufe_to_subgraph(
                 ):
                     node[key] = json.dumps(value, cls=JSON_HANDLER.encoder)
                     node["_json_props"].append(key)
+            elif isinstance(value, Settings):
+                # TODO: finish up approach here for serializing settings
```
Comment (hmacdope) [inline alchemiscale/storage/statestore.py:324]: Is this still TODO?

File: alchemiscale/storage/statestore.py [hunk 3]
```diff
@@ -1358,7 +1364,10 @@ def get_task_transformation(
         self,
         task: ScopedKey,
         return_gufe=True,
-    ) -> Tuple[Transformation, Optional[ProtocolDAGResultRef]]:
+    ) -> Union[
+        Tuple[Transformation, Optional[ProtocolDAGResultRef]],
```
Comment (hmacdope) [inline alchemiscale/storage/statestore.py:1368]: Given the two possible return types might be good to have a quick note in docstring as to under which circumstances each is returned

**Review by hmacdope — CHANGES_REQUESTED** [review]
Just a few nitpicks, but looking great! Thanks for fixing up.

File: alchemiscale/base/client.py (hunk 1 above)
Comment (dotsdl) [reply alchemiscale/base/client.py:127]: Yeah, we ended up disabling the behavior 90% of the time, so seemed right to just drop it.

File: alchemiscale/storage/statestore.py (hunk 2 above)
Comment (dotsdl) [reply alchemiscale/storage/statestore.py:324]: Nope! All accounted for. Thanks for catching this!

File: alchemiscale/storage/statestore.py (hunk 3 above)
Comment (dotsdl) [reply alchemiscale/storage/statestore.py:1366]: Done!

**Comment by dotsdl** [comment]
Comments addressed @hmacdope! Thanks for the review!

**Review by hmacdope — APPROVED** [review]
Thanks @dotsdl

**Pull request merged**